src/core/Core/Application.cpp

```diff
@@ -131,403 +198,203 @@ ExitStatus App::Application::run() {
+        ImGui::Begin("Right Pane",
+            nullptr,
+            ImGuiWindowFlags_NoResize | ImGuiWindowFlags_NoMove | ImGuiWindowFlags_NoCollapse |
+                ImGuiWindowFlags_NoTitleBar);
+
         ImDrawList* draw_list = ImGui::GetWindowDrawList();
-        const ImVec2 canvas_p0 = ImGui::GetCursorScreenPos();
-        const ImVec2 canvas_sz = ImGui::GetContentRegionAvail();
-        const auto canvas_p1 = ImVec2(canvas_p0.x + canvas_sz.x, canvas_p0.y + canvas_sz.y);
-        // --- PANNING IMPLEMENTATION BEGIN ---
+
+        // Define the graphing area within the window
+        const ImVec2 canvas_p0 = ImGui::GetCursorScreenPos();     // Top-left corner of the window
+        const ImVec2 canvas_sz = ImGui::GetContentRegionAvail();  // Size of the window
+        const auto canvas_p1 =
+            ImVec2(canvas_p0.x + canvas_sz.x, canvas_p0.y + canvas_sz.y);  // Bottom-right
+
+        // Define your coordinate system origin (e.g., center of the canvas)
+        const ImVec2 origin(canvas_p0.x + canvas_sz.x * 0.5f, canvas_p0.y + canvas_sz.y * 0.5f);
+
+        float lineThickness = 3.0f;
+
+        //panning logic
         static bool isPanning = false;
-        static ImVec2 lastMousePos = {0.0f, 0.0f};
-        static ImVec2 originOffset = {0.0f, 0.0f};
+        static ImVec2 panStart = ImVec2(0,0);
 
-        // Get mouse position
-        ImVec2 mousePos = ImGui::GetMousePos();
+        //get mouse position and state
+        ImGuiIO& io=ImGui::GetIO();
+        ImVec2 mousePos = io.MousePos;
 
-        // Detect click start only when cursor is inside the graphing area
-        if (ImGui::IsWindowHovered() && ImGui::IsMouseClicked(ImGuiMouseButton_Left)) {
+        // Check if mouse is over the canvas
+        bool isHovered = ImGui::IsWindowHovered();
+
+        // Right click pressed: start panning
+        if (isHovered && ImGui::IsMouseClicked(ImGuiMouseButton_Right)) {
             isPanning = true;
-            lastMousePos = mousePos;
+            panStart = mousePos;
         }
 
-        // Stop panning when mouse released
-        if (ImGui::IsMouseReleased(ImGuiMouseButton_Left)) {
+        // Mouse released: stop panning
+        if (ImGui::IsMouseReleased(ImGuiMouseButton_Right)) {
             isPanning = false;
         }
 
-        // While dragging, update origin offset
-        if (isPanning && ImGui::IsMouseDragging(ImGuiMouseButton_Left)) {
-            ImVec2 delta = ImVec2(mousePos.x - lastMousePos.x, mousePos.y - lastMousePos.y);
-            originOffset.x += delta.x;
-            originOffset.y += delta.y;
-            lastMousePos = mousePos;
+        // If panning, update offset
+        if (isPanning && ImGui::IsMouseDragging(ImGuiMouseButton_Right)) {
+            ImVec2 dragDelta = ImGui::GetMouseDragDelta(ImGuiMouseButton_Right);
+            offsetx += dragDelta.x;
+            offsety += dragDelta.y;
+            ImGui::ResetMouseDragDelta(ImGuiMouseButton_Right);
         }
```

Comment:
Please use the left mouse button instead of right for panning. If that is not possible, do let me know.